dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hdfs/ScanContext.java

```diff
@@ -141,38 +197,112 @@ void walkFile(FileStatus file) throws IOException {
           /* nFiles= */ 0,
           /* nDirs= */ 0,
           strStoragePolicy);
+
+      maybeLogStats(file);
     }
   }
 
-  /** This method is used to produce meaningful metrics for log/debug purposes. */
-  String getFormattedStats() {
+  @GuardedBy("csvPrinter")
+  private long lastTimeStatsLogged;
+
+  private void maybeLogStats(FileStatus file) {
+    long currentTime = System.currentTimeMillis();
+    if (currentTime - lastTimeStatsLogged >= PROGRESS_UPDATE_INTERVAL) {
```

Comment:
The idea of this logic is fine and really generic. Could you extract it into new abstraction, so we will be able to reuse it in other connectors as well? 

Something like logs\progress throttle.

Reply:
I think it also should help to write unit-tests for this dedicated component, because it will be less logic in it.